core/src/identity/ed25519.rs

```diff
@@ -138,7 +139,13 @@ impl Clone for SecretKey {
 impl SecretKey {
     /// Generate a new Ed25519 secret key.
     pub fn generate() -> SecretKey {
-        SecretKey(ed25519::SecretKey::generate(&mut rand::thread_rng()))
+        loop {
+            let mut bytes = [0u8; 32];
+            rand::thread_rng().fill_bytes(&mut bytes);
+            if let Ok(key) = ed25519::SecretKey::from_bytes(&bytes) {
```

Comment:
This can only `Err` if the array size is wrong. The right fix is to add `expect("<some proof>")`.